core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionPolicyManager.java

```diff
@@ -46,8 +50,16 @@
  *
  * Created by gupele on 6/29/2015.
  */
-public final class TransmissionPolicyManager implements Stoppable {
-
+public final class TransmissionPolicyManager implements Stoppable, TransmissionHandlerObserver {
+
+	private final int DEFAULT_MAX_SECONDS_TO_PAUSE_AFTER_MAX_BACKOFF = 600;
```

Comment:
How did we came up with this number ? 600 Seconds seems to be too long. In a situation where the network is very restricted this would accumulate bulk of telemetry.

Reply:
Agreed, 10 minutes is a long time and it should be configurable. In the .NET implementation it will max out at 30 minutes in the backoff algorithm. Let's talk about this a bit more to see what we really want to do. By the time we reach this point we would have executed the backoff 15 times and probably have been throttled well over 15 minutes. It's possible this is just redundant.

Reply:
@debugthings Yes, ideally I would want that to be configurable too. So instead of keeping that 15 constants for back of we have some sort of iterator pattern implemented which keeps on providing new backoff numbers until user specified limit. Internally iterator could well be implemented as Geometric Progression with ratio of 2. If we want to interleave with a constant time that could also be done.